src/test/clojure/xtdb/trie_catalog_test.clj

```diff
@@ -101,3 +112,55 @@
         (t/is (= #{"l00-rc-b01" "l00-rc-b00"}
                  (->> (cat/current-tries (cat/trie-state cat "public/foo"))
                       (into #{} (map :trie-key)))))))))
+
+(defn added-trie->edn [^AddedTrie trie]
+  {:table-name (.getTableName trie)
+   :trie-key (.getTrieKey trie)
+   :data-file-size (.getDataFileSize trie)})
+
+(t/deftest current-tries-on-finish-block
+  (let [node-dir (util/->path "target/trie-catalog-test/current-tries-on-finish-block")]
+    (util/delete-dir node-dir)
+
+    (with-open [node (tu/->local-node {:node-dir node-dir, :compactor-threads 0})]
+      (let [^BufferPool bp (tu/component node :xtdb/buffer-pool)
+            ^TrieCatalog trie-catalog (tu/component node :xtdb/trie-catalog)]
+        (xt/execute-tx node [[:put-docs :foo {:xt/id 1}]])
+        (tu/finish-block! node)
+
+        (xt/execute-tx node [[:put-docs :foo {:xt/id 2}]])
+        (tu/finish-block! node)
+
+        (t/is (= [(os/->StoredObject "tables/public$foo/blocks/b00.binpb" 186)
+                  (os/->StoredObject "tables/public$foo/blocks/b01.binpb" 215)]
+                 (.listAllObjects bp (trie/->table-block-dir "public/foo"))))
+
+        (t/is (= [{:table-name "public/foo",
+                   :trie-key "l00-rc-b00",
+                   :data-file-size 1966}
+                  {:table-name "public/foo",
+                   :trie-key "l00-rc-b01",
+                   :data-file-size 1966}]
+                 (->> (.getByteArray bp (util/->path "tables/public$foo/blocks/b01.binpb"))
+                      TableBlock/parseFrom
+                      trie/<-table-block
+                      :current-tries
+                      (mapv added-trie->edn ))))
+
+        (t/testing "artifically adding tries (simulating another node finishing and compacting these)"
+          (->> [["l00-rc-b00" 1] ["l00-rc-b01" 1] ["l00-rc-b02" 1] ["l00-rc-b03" 1]
+                ["l01-rc-b00" 2] ["l01-rc-b01" 2] ["l01-rc-b02" 2]
+                ["l02-rc-p0-b01" 4] ["l02-rc-p1-b01" 4] ["l02-rc-p2-b01" 4] ["l02-rc-p3-b01"4]]
+               (map #(apply (partial cat/->added-trie "public/foo") %))
+               (.addTries trie-catalog))
+
+          (tu/finish-block! node)
+
+          (t/is (= ["l02-rc-p0-b01" "l02-rc-p1-b01" "l02-rc-p2-b01" "l02-rc-p3-b01"
+                    "l01-rc-b02"
+                    "l00-rc-b03"]
+                   (->> (.getByteArray bp (util/->path "tables/public$foo/blocks/b02.binpb"))
+                        TableBlock/parseFrom
+                        trie/<-table-block
+                        :current-tries
+                        (mapv (comp :trie-key added-trie->edn))))))))))
```

Comment:
This last part of the test is kind of weird as I am artificially adding trie notifications.